www/tests/test_app.py

```diff
@@ -1,4 +1,11 @@
 import app
 
-def test_dummy():
+def test_sensor_init():
     assert len(app.binary_sensors) > 0
+
+def test_get_local_ip():
```

Comment:
Maybe a better way is to add an optional parameter (`run_once`) that is `false` by default, but that the test can set to true so that it just runs through once. That won't give 100% test coverage, but it will test most of it. I think you'd have to spin up a task in your test in order to stop the task in the infinite loop case.